consul/autopilot.go

```diff
@@ -98,13 +118,17 @@ func (s *Server) pruneDeadServers() error {
 	}
 
 	// Only do removals if a minority of servers will be affected
-	if len(failed) < peers/2 {
+	if removalCount < peers/2 {
 		for _, server := range failed {
 			s.logger.Printf("[INFO] consul: Attempting removal of failed server: %v", server)
 			go s.serfLAN.RemoveFailedNode(server)
 		}
+		for _, raftServer := range staleRaftServers {
+			s.logger.Printf("[INFO] consul: Attempting removal of stale raft server : %v", raftServer.ID)
+			s.raft.RemoveServer(raftServer.ID, 0, 0)
```

Comment:
Do we need this logic here? https://github.com/hashicorp/consul/blob/master/consul/leader.go#L687-L706

Might be good to put that into a helper fn.